module-apps/apps-common/notifications/NotificationTilesModel.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "NotificationTilesModel.hpp"
+#include <Image.hpp>
+#include <Style.hpp>
+
+using namespace gui;
+
+namespace
+{
+    std::map<notifications::NotificationType, UTF8> typeToIcon{
+        {notifications::NotificationType::NotSeenSms, "messages_notification_icon_W_G"},
+        {notifications::NotificationType::NotSeenCall, "calls_notification_icon_W_G"},
+        {notifications::NotificationType::Tethering, "tethering_notification_icon_W_G"},
+        {notifications::NotificationType::AlarmSnooze, "alarm_notification_icon_W_G"},
+        {notifications::NotificationType::PhoneLock, "lock_notification_icon_W_G"}};
+    auto buildNotificationIcon(const UTF8 &imageName) -> gui::Image *
+    {
+        auto icon = new gui::Image(imageName);
+        icon->setMinimumWidth(::style::wallpaper::notificationTiles::tileIconHeight);
+        icon->setMargins(gui::Margins(::style::wallpaper::notificationTiles::tileIconMarigin,
+                                      0,
+                                      ::style::wallpaper::notificationTiles::tileIconMarigin,
+                                      0));
+        return icon;
+    }
+} // namespace
+
+NotificationTilesModel::NotificationTilesModel()
+{}
+
+void NotificationTilesModel::attach(HBox *tilesBox)
+{
+    this->tilesBox = tilesBox;
+}
+
+void NotificationTilesModel::updateData(app::manager::actions::NotificationsChangedParams *params,
+                                        bool callAndSMSVisibility)
+{
+    if (tilesBox == nullptr) {
+        LOG_ERROR("TilesBox not attached");
+        return;
+    }
+
+    tilesBox->erase();
+
+    for (const auto &notification : params->getNotifications()) {
+        if (typeid(*notification) == typeid(notifications::NotSeenSMSNotification) && callAndSMSVisibility) {
```

Comment:
I don't think it's MVP (or MVC to be honest :) ) now 
good crib: [link](http://1.bp.blogspot.com/-8751o9FiHdE/Vp1V7xzOz7I/AAAAAAAAAtk/GAfrW9lvT2U/s1600/mvp.png)

Reply:
I know that it may not be clear at first glance but I had to make some tradeoffs to achieve:
- one notifications model to support multiple views (both listView and tiles)
- one tiles notification model to drive multiple instances of NotificationTilesBox (HBox) - switchable dynamically

Reply:
Renamed to provide MVP structure:
model (NotificationsModel) <> presenter (Notifications tiles/list presenter) <> View (ListView, NotificationTiles)